File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> temp;
        vector<vector<string>> ans;
        for( string s: strs){
            vector<int> fr(26,0);
            for( char c: s){
                fr[c-'a']++;
            }
            string tem ="";
            for(int i : fr) tem= tem+to_string(i)+" ";
            if(temp.find(tem)==temp.end()) temp[tem]={s};
            else temp[tem].push_back(s);
        }
        for(auto i = temp.begin();i!=temp.end();i++) ans.push_back(i->second);
        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> mp = new HashMap<>();
        List<List<String>> ans = new ArrayList<>();
        for( String s : strs)
        {
             int[] freq = new int[26];
            for(Character ch : s.toCharArray())
            {
                freq[ch-'a']++;
            }
            StringBuilder sb= new StringBuilder("");
            for(int i=0;i<26;i++)
            {
               sb = sb.append( i+","+freq[i]+",");
            }
            if(mp.get(sb.toString())==null){
                 mp.put(sb.toString(),new ArrayList<>());
            }
            mp.get(sb.toString()).add(s);

        }
        
        return new ArrayList<>(mp.values());
    }
}